libr/util/alloc.c

```diff
@@ -0,0 +1,51 @@
+/* radare - LGPL - Copyright 2018 - pancake */
+
+#include <r_util/r_alloc.h>
+
+#if R_MALLOC_WRAPPER
+
+#if R_MALLOC_GLOBAL
+R_API void r_alloc_hooks(RMalloc m, RCalloc c, RRealloc r, RFree f) {
+	r_malloc = m? m: malloc;
+	r_calloc = c? c: calloc;
+	r_realloc = r? r: realloc;
+	r_free = f? f: free;
+}
+
+#else
+static RMalloc *_r_malloc = malloc;
```

Comment:
can't you just reuse the global pointers that you use when `R_MALLOC_GLOBAL`?

Reply:
uh?

Reply:
maybe we should just assert because there's no sense in  just setting some o f thoe pointers and not others